### DISCUSSION TITLE
[Bug]: NEXTCLOUD_ADDITIONAL_APKS NEXTCLOUD_ADDITIONAL_PHP_EXTENSIONS not working

### DISCUSSION THREAD

**Discussion by FlattusBlastus** [nextcloud/all-in-one#1856] [Questions] [answered]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

I have tried multiple attempts to add modules via docker run into Nextcloud AIO per the instructions. All single module adds fail with no error message. Multiple modules give an error message that access to the repository is denied and their is no local version of the module.

### Steps to reproduce

1. Use -e NEXTCLOUD_ADDITIONAL_APKS=libpq-dev imagemagick ^
-e NEXTCLOUD_ADDITIONAL_PHP_EXTENSIONS=pgsql imagick ^
or any combo in your docker run for the mastercontainer.

2 Observe the error message.

### Expected behavior

The modules are supposed to be fetched and automatically installed.

### Installation method

Official All-in-One appliance

### Operating system

None

### PHP engine version

None

### Web server

Apache (supported)

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
Straight after AIO module load - stopped here
```

### List of activated Apps

```shell
Straight after AIO module load - stopped here
```

**Comment by Zoey2936** [comment]
can you please send the entire docker run command?
and what ist the output of: `docker inspect nextcloud-aio-mastercontainer`

**Comment by FlattusBlastus** [comment]
docker run ^
--sig-proxy=false ^
--name nextcloud-aio-mastercontainer ^
--restart always ^
--publish 8080:8080 ^
-e APACHE_PORT=11000 ^
-e NEXTCLOUD_DATADIR="nextcloud_aio_nextcloud_datadir" ^
-e SKIP_DOMAIN_VALIDATION=true ^
-e NEXTCLOUD_MEMORY_LIMIT=4096M ^
-e NEXTCLOUD_ADDITIONAL_APKS=libpq-dev imagemagick ^
-e NEXTCLOUD_ADDITIONAL_PHP_EXTENSIONS=pgsql imagick ^
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config ^
--volume //var/run/docker.sock:/var/run/docker.sock:ro ^
nextcloud/all-in-one:latest

**Reply by Zoey2936** [reply]
can you try to quote the envs? (NEXTCLOUD_ADDITIONAL_APKS="libpq-dev imagemagick")

**Comment by FlattusBlastus** [comment]
[
    {
        "Id": "4c5f6b5482f116109673c48a5a3db5c82cf590b9d6a5eb0ad13aad3d86ec8946",
        "Created": "2023-01-29T01:00:36.413162545Z",
        "Path": "start.sh",
        "Args": [
            "/usr/bin/supervisord",
            "-c",
            "/supervisord.conf"
        ],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 26941,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2023-01-29T01:00:36.783459697Z",
            "FinishedAt": "0001-01-01T00:00:00Z",
            "Health": {
                "Status": "healthy",
                "FailingStreak": 0,
                "Log": [
                    {
                        "Start": "2023-01-30T12:43:20.017939206Z",
                        "End": "2023-01-30T12:43:20.144559853Z",
                        "ExitCode": 0,
                        "Output": ""
                    },
                    {
                        "Start": "2023-01-30T12:43:50.148423646Z",
                        "End": "2023-01-30T12:43:50.266742329Z",
                        "ExitCode": 0,
                        "Output": ""
                    },
                    {
                        "Start": "2023-01-30T12:44:20.27028574Z",
                        "End": "2023-01-30T12:44:20.357715089Z",
                        "ExitCode": 0,
                        "Output": ""
                    },
                    {
                        "Start": "2023-01-30T12:44:50.361737907Z",
                        "End": "2023-01-30T12:44:50.48437528Z",
                        "ExitCode": 0,
                        "Output": ""
                    },
                    {
                        "Start": "2023-01-30T12:45:20.490165153Z",
                        "End": "2023-01-30T12:45:20.594970951Z",
                        "ExitCode": 0,
                        "Output": ""
                    }
                ]
            }
        },
        "Image": "sha256:35049b54ac64e0e2d11caf945600199fcc363eb42b5a2404c05b0b6a848d8f97",
        "ResolvConfPath": "/var/lib/docker/containers/4c5f6b5482f116109673c48a5a3db5c82cf590b9d6a5eb0ad13aad3d86ec8946/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/4c5f6b5482f116109673c48a5a3db5c82cf590b9d6a5eb0ad13aad3d86ec8946/hostname",
        "HostsPath": "/var/lib/docker/containers/4c5f6b5482f116109673c48a5a3db5c82cf590b9d6a5eb0ad13aad3d86ec8946/hosts",
        "LogPath": "/var/lib/docker/containers/4c5f6b5482f116109673c48a5a3db5c82cf590b9d6a5eb0ad13aad3d86ec8946/4c5f6b5482f116109673c48a5a3db5c82cf590b9d6a5eb0ad13aad3d86ec8946-json.log",
        "Name": "/nextcloud-aio-mastercontainer",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": [
                "nextcloud_aio_mastercontainer:/mnt/docker-aio-config",
                "/var/run/docker.sock:/var/run/docker.sock:ro"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "default",
            "PortBindings": {
                "8080/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "8080"
                    }
                ]
            },
            "RestartPolicy": {
                "Name": "always",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "CgroupnsMode": "host",
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                30,
                120
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": [],
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [],
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "KernelMemory": 0,
            "KernelMemoryTCP": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/a06dfb3d3c2ceda11150e5ffb803db013089ae3eac5090b19fa7ed7f9c17fa2f-init/diff:/var/lib/docker/overlay2/6ae2a685f66c93b36909d8243e61fa2d04be118097534e7f3f95d83679c443e7/diff:/var/lib/docker/overlay2/b53a05fc948eb1f1e075f8a673888b795122318c7fa8bf767bd1cdefadb598d3/diff:/var/lib/docker/overlay2/1f0668381bf3cd414a9333d50fb583c8345ad0961b32ebeb299dcaa0fc76bd24/diff:/var/lib/docker/overlay2/26deb8b36db33aff7cc8d091fc85047b2f64ccbd489b22970dea971bb4aff10b/diff:/var/lib/docker/overlay2/be16db4dd0de6dbf8563c99ba192dbfc9b8d9101e86a3fa2cab29d299e65b6a3/diff:/var/lib/docker/overlay2/18eb72aee0ef5f18ccb7673e5ee89e72215b7025d33383782413b8a875a606ff/diff:/var/lib/docker/overlay2/55662ce39c4246279de9fabcc6d73806e4486aba8ba28696e47a0bca74d73ac9/diff:/var/lib/docker/overlay2/ca2792342ba088271eb543d7843349b301e0a24e7cf0f2fbc02d598ffd9ae5ae/diff:/var/lib/docker/overlay2/a40d8f13c001b3ddc3b9f3274c7fec2a46b06ca87a1a00975bc0504b4c9062fe/diff:/var/lib/docker/overlay2/db182b01c363f998030a54c0f52fb29dfa4a6c65d6b539e37bf7d934538fb3e0/diff:/var/lib/docker/overlay2/50e108f2fb71d593c07ac7c635d5b0f74dd15803458435843ecafa1a30de9d1c/diff:/var/lib/docker/overlay2/e7a9c9c267278eb2527fa1a508704ab7d9390e64c0dd718f5b157a5f5b6727fb/diff:/var/lib/docker/overlay2/4f95d3371e873d87c514ebb683427414ef94debb480a986cfb6384b3727dd529/diff:/var/lib/docker/overlay2/fc9b3b6a6c8b39cbf64ab6fdcf72c59c096209f8472f2be825bef366fdb02097/diff:/var/lib/docker/overlay2/5bfc91fa017cb036700c5167b0765fe589a2d2cddaea65572c36073323ac635c/diff:/var/lib/docker/overlay2/025df53cf39318b8c08950cab581b44abf352d15e28173e42b40162fb5e4c412/diff:/var/lib/docker/overlay2/50b9cc0c05bf8d71f9742f77ec034259acf78081cf6c698b149b44b18bfe0026/diff:/var/lib/docker/overlay2/e30c1fc409d6d8b837c71e2d8f8db8d305f9eab205ac348cbfc5f0740425bd38/diff:/var/lib/docker/overlay2/e02889c182a5853929de39d4fb107dec7908b744746b504fb508899806f02c18/diff:/var/lib/docker/overlay2/b77afae775106ba3e9e92be8fd480c409003b817d1468f0afbe67a273f1ea92e/diff:/var/lib/docker/overlay2/a67ebb5ad41d95c1f4aebfa2804a6635fc60cf8d6e8f0740f4714a8310e8b36b/diff:/var/lib/docker/overlay2/3bbcd0dc20fff743a7c165e2041619dec1a6dccba2cd774c31202639059f3851/diff:/var/lib/docker/overlay2/b28043639263ec256733a9f8218108ba1c00351a11ffc0132aff1c1a538a9633/diff:/var/lib/docker/overlay2/26ad90fe89d12c627f3cd329fddb2ebe621348ce3c7395d448770e0707c492b2/diff:/var/lib/docker/overlay2/0383af0fcdc8ae4eb819478cca804d772cbaf949ce756cf0caf28f36e2ae6b1c/diff:/var/lib/docker/overlay2/a3fecbf2228d7d5d93d7c534f6b785fe06586731b82e28c7e115726236b947ae/diff:/var/lib/docker/overlay2/62cf90c12430a94c6dbec5092c92daa5ee73d8a14f4cbc767e87b03fa487d71f/diff:/var/lib/docker/overlay2/049abccf710b1771c8306b342d7579b75d79f3b1af603135326b659c2b1e4abb/diff:/var/lib/docker/overlay2/55a55170d004d3fa711da0e642ea76c212b31567f74f54dce763b2630ab3f30f/diff:/var/lib/docker/overlay2/3622d94be7923a870488a17da99bc4eb89d6d4a9116575c39585b2c950959fe3/diff:/var/lib/docker/overlay2/4496b1276950a2102fc3ca7a16e8b16bf46bdb9759826b0bb6bc8d4ff61c0d06/diff:/var/lib/docker/overlay2/415a811b9c4e58c72b59a06aa09b187831104f35b3939b32e00e169032098c60/diff:/var/lib/docker/overlay2/4f7eebc49fe58ede37555ce9b8fc4acf0404a4b6635393928763e85f77118253/diff:/var/lib/docker/overlay2/3353bbec8fa9bff248bb22a8de56c5cf29d0ca1279172a1b3d9b11e422efd092/diff:/var/lib/docker/overlay2/2dd416fd06d93cf4302d4469eb80054e7350adf6a8cd0bd73c15d2346ecf71a3/diff:/var/lib/docker/overlay2/023764a3adaec5ede954ba40463ef50df94e7fff6213465af30a9a1089754ece/diff:/var/lib/docker/overlay2/6dee0b4fc0a12c839cc5c2c8847af051793a62b3eb5d1aa43b114b45586c0473/diff:/var/lib/docker/overlay2/9d99d5885a77fa11f20b767407c76ca5c1f0697bde8bdea5cca86d49058e579b/diff",
                "MergedDir": "/var/lib/docker/overlay2/a06dfb3d3c2ceda11150e5ffb803db013089ae3eac5090b19fa7ed7f9c17fa2f/merged",
                "UpperDir": "/var/lib/docker/overlay2/a06dfb3d3c2ceda11150e5ffb803db013089ae3eac5090b19fa7ed7f9c17fa2f/diff",
                "WorkDir": "/var/lib/docker/overlay2/a06dfb3d3c2ceda11150e5ffb803db013089ae3eac5090b19fa7ed7f9c17fa2f/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [
            {
                "Type": "volume",
                "Name": "nextcloud_aio_mastercontainer",
                "Source": "/var/lib/docker/volumes/nextcloud_aio_mastercontainer/_data",
                "Destination": "/mnt/docker-aio-config",
                "Driver": "local",
                "Mode": "z",
                "RW": true,
                "Propagation": ""
            },
            {
                "Type": "bind",
                "Source": "/var/run/docker.sock",
                "Destination": "/var/run/docker.sock",
                "Mode": "ro",
                "RW": false,
                "Propagation": "rprivate"
            }
        ],
        "Config": {
            "Hostname": "4c5f6b5482f1",
            "Domainname": "",
            "User": "root",
            "AttachStdin": false,
            "AttachStdout": true,
            "AttachStderr": true,
            "ExposedPorts": {
                "80/tcp": {},
                "8080/tcp": {},
                "8443/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "APACHE_PORT=11000",
                "NEXTCLOUD_DATADIR=nextcloud_aio_nextcloud_datadir",
                "SKIP_DOMAIN_VALIDATION=true",
                "NEXTCLOUD_MEMORY_LIMIT=4096M",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "PHPIZE_DEPS=autoconf \t\tdpkg-dev \t\tfile \t\tg++ \t\tgcc \t\tlibc-dev \t\tmake \t\tpkg-config \t\tre2c",
                "PHP_INI_DIR=/usr/local/etc/php",
                "APACHE_CONFDIR=/etc/apache2",
                "APACHE_ENVVARS=/etc/apache2/envvars",
                "PHP_CFLAGS=-fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64",
                "PHP_CPPFLAGS=-fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64",
                "PHP_LDFLAGS=-Wl,-O1 -pie",
                "GPG_KEYS=528995BFEDFBA7191D46839EF9BA0ADA31CBD89E 39B641343D8C104B2B146DC3F9C39DC0B9698544 F1F692238FBC1666E5A5CCD4199F9DFEF6FFBAFD",
                "PHP_VERSION=8.1.14",
                "PHP_URL=https://www.php.net/distributions/php-8.1.14.tar.xz",
                "PHP_ASC_URL=https://www.php.net/distributions/php-8.1.14.tar.xz.asc",
                "PHP_SHA256=e16e47a872d58685913ac848ce92ec49f42c1828110c98c65fb6265a08724a1a"
            ],
            "Cmd": [
                "/usr/bin/supervisord",
                "-c",
                "/supervisord.conf"
            ],
            "Healthcheck": {
                "Test": [
                    "CMD-SHELL",
                    "/healthcheck.sh"
                ]
            },
            "Image": "nextcloud/all-in-one:latest",
            "Volumes": {
                "/mnt/docker-aio-config/": {}
            },
            "WorkingDir": "/var/www/docker-aio",
            "Entrypoint": [
                "start.sh"
            ],
            "OnBuild": null,
            "Labels": {},
            "StopSignal": "SIGWINCH"
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "46fed6ceeaf00f24b30a5aff8ee19b29968a6fd8afaaef0916856455e98c60ef",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "80/tcp": null,
                "8080/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "8080"
                    }
                ],
                "8443/tcp": null
            },
            "SandboxKey": "/var/run/docker/netns/46fed6ceeaf0",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "2c472f94493449b16f1afaed4c419a89883d81448975454f2be09fb192fc609b",
            "Gateway": "172.17.0.1",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "172.17.0.2",
            "IPPrefixLen": 16,
            "IPv6Gateway": "",
            "MacAddress": "02:42:ac:11:00:02",
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "4490135bbd190d231a12178be9cf3eddf9585c8799a2668f6b2289c2f9bf96f8",
                    "EndpointID": "2c472f94493449b16f1afaed4c419a89883d81448975454f2be09fb192fc609b",
                    "Gateway": "172.17.0.1",
                    "IPAddress": "172.17.0.2",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:ac:11:00:02",
                    "DriverOpts": null
                },
                "nextcloud-aio": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "4c5f6b5482f1"
                    ],
                    "NetworkID": "b15b6e1f3436365005e3561742f2e936e0ca20275c06c96fd622d005b097c4b7",
                    "EndpointID": "ee00b09e39920c5751dcde6aa4d9d224c9f5a72d01dc19e70eb2bf5043105dbf",
                    "Gateway": "172.18.0.1",
                    "IPAddress": "172.18.0.11",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:ac:12:00:0b",
                    "DriverOpts": null
                }
            }
        }
    }
]

**Comment by szaimen** [comment, marked as answer]
> can you try to quote the envs? (NEXTCLOUD_ADDITIONAL_APKS="libpq-dev imagemagick")
